xdsl/parser.py

```diff
@@ -584,6 +584,12 @@ def parse_optional_xdsl_builtin_attribute(self,
                 typ = Float32Type()
             return FloatAttr.from_value(float_lit, typ)
 
+        # Shorthand for boolean literals (IntegerAttr of width 1)
```

Comment:
Maybe pack those into a single function? Otherwise, you are repeating the exact same code multiple times...